pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -121,36 +121,31 @@ let
       src = fetchurl { inherit urls sha512; };
 
       passthru = {
-        inherit pname tlType version;
-      } // lib.optionalAttrs (sha512 != "") { inherit src; };
+        inherit pname src tlType version;
+      };
       unpackCmd = file: ''
         tar -xf ${file} \
           '--strip-components=${toString stripPrefix}' \
           -C "$out" --anchored --exclude=tlpkg --keep-old-files
       '' + postUnpack;
 
-    in if sha512 == "" then
-      # hash stripped from pkgs.nix to save space -> fetch&unpack in a single step
+    in if fixedHash != null then
+      # hash already computed -> fetch&unpack with a single derivation to save space
       fetchurl {
-        inherit urls;
-        sha1 = if fixedHash == null then throw "TeX Live package ${tlName} is missing hash!"
-          else fixedHash;
-        name = tlName;
+        inherit passthru urls;
+        sha1 = fixedHash;
+        name = "texlive-${tlName}";
```

Comment:
I now understand that the `texlive-` prefix is just to help `fixHashes.awk` do the work, so it shouldn't be added here. I will remove it before the final squash.

Reply:
We can keep add the prefix if it looks fine.

Reply:
I actually like the different prefixes, it reveals which code path is being followed. If anything, I would use a more obvious prefix in the other derivation, like `texlivenonfdo-`. Not for this PR of course.